sklearn/utils/estimator_checks.py

```diff
@@ -137,12 +137,18 @@ def _yield_checks(estimator):
     yield check_estimator_get_tags_default_keys
 
     if tags["array_api_support"]:
-        for array_namespace in ["numpy.array_api", "cupy.array_api"]:
-            yield partial(check_array_api_input, array_namespace=array_namespace)
-
-        for device in ["cpu", "cuda"]:
-            for dtype in ("float64", "float32"):
-                yield partial(check_array_api_input_torch, dtype=dtype, device=device)
+        for array_namespace in ["numpy.array_api", "cupy.array_api", "torch"]:
+            if array_namespace == "torch":
+                for device in ["cpu", "cuda"]:
+                    for dtype in ("float64", "float32"):
+                        yield partial(
+                            check_array_api_input,
+                            array_namespace=array_namespace,
+                            dtype=dtype,
+                            device=device,
+                        )
+            else:
+                yield partial(check_array_api_input, array_namespace=array_namespace)
```

Comment:
You should pass `device="gpu"` for `cupy` or alternatively `device=None` might work for automated device selection?

Reply:
I went ahead and changed the default to `device=None` to fix the tests under cupy.